### ISSUE TITLE
Maybe migrate tests to mocha

### ISSUE THREAD

**Issue by johndigital** [funkhaus/fh-components#20]
There is some discussion about this in https://github.com/funkhaus/fh-components/issues/19

Look into mocha vs. tap and figure out which will be better for us to use for testing moving forward.

Most of the examples in the [vue test library documentation](https://vue-test-utils.vuejs.org/en/guides/getting-started.html) are written in mocha syntax.

**Comment by SaFrMo** [comment]
Looks like Vue [recommends Karma](https://vuejs.org/v2/guide/unit-testing.html) - not sure what the difference is, but I noticed this while going through the docs!

**Comment by johndigital** [comment]
After doing some research into this I ended up deciding we should go with mocha. That's not to say we won't use Karma at some point, apparently Karma is a higher level wrapper that tests things in real browsers and works really well alongside mocha. But I don't think we need all the features of karma right now.

I've re-written the existing tests using mocha syntax and it works really well, the error outputting is much better IMO than what we had before with tap. The documentation is also really nice and it goes really deep with advanced functionality. 

It's also agnostic of what assertion library you use, which is another advantage. I've used just the regular node assert in the re-write but it might be worth looking through some of the [popular assertion options](https://mochajs.org/#assertions) and picking one we like.

One last note is that a big difference as far as I can tell between tap and mocha is that the individual tests do not necessarily run in a particular order. That means each test should handle its own setup and teardown and not depend on any other test, which is just good practice in general but mocha sort of forces it.
